radicle-cli/src/commands/merge.rs

```diff
@@ -144,10 +141,27 @@ pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
     //
     // Get patch information
     //
-    let patch_id = options.id.resolve(&repository.backend)?;
+    let revision_id = options.revision_id.resolve(&repository.backend)?;
+    let (patch_id, patch) = patches.find_by_revision(&revision_id)?.ok_or(anyhow!(
+        "no open patch with revision `{}` could be found",
+        &revision_id
+    ))?;
+    if !patch.is_open() {
+        anyhow::bail!(
+            "revision's patch must be open for merging, but it is `{}`",
+            patch.state()
+        );
+    }
+    let (last_revision_id, _) = patch.latest().expect("patch has one or more revisions");
+    if !options.force && revision_id != *last_revision_id {
+        anyhow::bail!("refusing to merge old patch revision");
+    }
+
     let mut patch = patches
         .get_mut(&patch_id)
-        .map_err(|e| anyhow!("couldn't find patch {} locally: {e}", &options.id.clone()))?;
+        .map_err(|e| anyhow!("couldn't find patch {} locally: {e}", &id))?;
+    // SAFETY: The patch is fetched by its revision_id above, and therefore expected to exist.
+    let revision = patch.revision(&revision_id).unwrap();
```

Comment:
Seems like it would make more sense if `find_by_revision` also returned the revision?